subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -599,20 +599,50 @@ static void api_event_handler(const struct nrf_cloud_evt *nrf_cloud_evt)
 static int api_init(const struct cloud_backend *const backend,
 		cloud_evt_handler_t handler)
 {
-	const struct nrf_cloud_init_param params = {
+	int ret;
+	struct nrf_cloud_init_param params = {
 		.event_handler = api_event_handler
 	};
 
+#if defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_RUNTIME)
+	if (!backend->config->id || !backend->config->id_len) {
+		LOG_ERR("ID has not been set in the backend config");
+		return -EINVAL;
+	} else if (backend->config->id_len > NRF_CLOUD_CLIENT_ID_MAX_LEN) {
+		LOG_ERR("ID must not exceed %d characters",
+			NRF_CLOUD_CLIENT_ID_MAX_LEN);
+		return -EBADR;
+	}
+
+	params.client_id = backend->config->id;
+#endif
+
 	backend->config->handler = handler;
 	nrf_cloud_backend = (struct cloud_backend *)backend;
 
-	return nrf_cloud_init(&params);
+	ret = nrf_cloud_init(&params);
+
+#if defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_RUNTIME)
+	nrf_cloud_free(backend->config->user_data);
```

Comment:
The `user_data` part does not seem to be needed anymore since we now use the `id` field?